### PR TITLE
chore: add cypress

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes:

- Added Cypress
- Added tests for basic functionality checks
Tests added for:
- [x] Search field
- [x] Toggle dark/light mode
- [x] Previous and Next button works
- [x] Sidebar works

## Context:

- Ref : #101

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
I like to have a real working test instead of so many sample files. Can we remove all non-used files please.

Also all files should end with `.ts`

https://docs.cypress.io/guides/tooling/typescript-support#Types-for-plugins

**Comment by RahulGautamSingh** [comment]
> I like to have a real working test instead of so many sample files. Can we remove all non-used files please.
> 
> Also all files should end with `.ts`
> 
> https://docs.cypress.io/guides/tooling/typescript-support#Types-for-plugins

WIP

**Comment by viceice** [comment]
CI should suceed after main merge 😉

**Comment by viceice** [comment]
Why you opened a new PR instead of reusing the old?
- #150

**Comment by RahulGautamSingh** [comment]
> Why you opened a new PR instead of reusing the old?
> 
> * [chore: add cypress #150](https://github.com/renovatebot/renovatebot.github.io/pull/150)

I was unable to resolve some conflicts after accidentally pushing a commit from WSL.

**Comment by viceice** [comment]
you can simply create a new commit and force-push, then all old commits are gone 😉

**Comment by HonkingGoose** [comment]
I use these Git commands often to clean up my work:

- `git commit --amend` to fix-up my last commit's content or commit message
- `git rebase -i` to edit, squash or drop commits
- `git push --force-with-lease` to force push my changes

I recommend the https://git-rebase.io/ website, it has a tutorial that explains how you can use all those commands. 😉

**Comment by RahulGautamSingh** [comment]
Removed the fixtures folder. 
Kept the support folder as it will come in use when we need testing library commands.

File: cypress/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "compilerOptions": {
+    "target": "es5",
```
Comment (viceice) [inline cypress/tsconfig.json:3]: I think we can use a more modern target? `es2020` should work, we can depend on node v16 if needed

File: cypress/plugins/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// cypress/plugins/index.ts
+
+/// <reference types="cypress" />
```
Comment (viceice) [inline cypress/plugins/index.ts:3]: I think this can be removed, it's already referenced in tsconfig

**Review by viceice — COMMENTED** [review]
Looks good so far, so please continue with more changes.

**Comment by rarkins** [comment]
@bahmutov are we doing this right? :)

**Comment by RahulGautamSingh** [comment]
@HonkingGoose I have added the basic set of tests as mentioned in #101. We will need more tests for other functionalities which can be added gradually. We can create a new issue where we can add tests to the list to it. Will make it easier to track the additions?

File: cypress/integration/sample.spec.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+describe('My First Test', () => {
+  beforeEach(() => {
+    cy.visit('http://127.0.0.1:8000/'); //change link when testing locally
```
Comment (viceice) [inline cypress/integration/sample.spec.ts:3]: where does cypress know to read our source files of? 🤔

can you add a CI workflow to have those basic tests running on CI ?

**Review by viceice — COMMENTED** [review]
LGTM so far

File: cypress/integration/sample.spec.ts (hunk 3 above)
Comment (RahulGautamSingh) [reply cypress/integration/sample.spec.ts:3]: Currently, it doesn't read source files.
I ran the docs page locally and cypress ran tests against it.
For CI we can point it to `https://docs.renovatebot.com`, I will work on creating a workflow 🤞

Comment (RahulGautamSingh) [reply cypress/integration/sample.spec.ts:3]: Can't use `https://docs.renovatebot.com` as we need to test updates as well. Will need to serve and use localhost addr

**Comment by RahulGautamSingh** [comment]
Added the workflow for CI but it fails on Post Set Up Python, not sure why??
 Need some help understanding the cause of it @viceice .

File: .github/workflows/cypress-tests.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        shell: bash
+        run: |
+          auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+          git submodule sync --recursive
+          git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
+
+      - name: Set up Node.js ${{ env.node-version }}
+        uses: actions/setup-node@9ced9a43a244f3ac94f13bfd896db8c8f30da67a # tag=v3.0.0
+        with:
+          node-version: ${{ env.node-version }}
```
Comment (viceice) [inline .github/workflows/cypress-tests.yml:27]: you forget to configure these env values.

**Review by viceice — APPROVED** [review]
LGTM, we should refactor workflows in a next PR, so we only deploy if all tests pass.

**Pull request merged**